hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/io/ECBlockReconstructedInputStream.java

```diff
@@ -0,0 +1,205 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.ozone.client.io;
+
+import org.apache.hadoop.hdds.client.BlockID;
+import org.apache.hadoop.hdds.client.ECReplicationConfig;
+import org.apache.hadoop.hdds.scm.storage.BlockExtendedInputStream;
+import org.apache.hadoop.hdds.scm.storage.ByteReaderStrategy;
+
+import java.io.EOFException;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+
+/**
+ * Input stream which wraps a ECBlockReconstructedStripeInputStream to allow
+ * a EC Block to be read via the traditional InputStream read methods.
+ */
+public class ECBlockReconstructedInputStream extends BlockExtendedInputStream {
+
+  private ECReplicationConfig repConfig;
+  private ECBlockReconstructedStripeInputStream stripeReader;
+  private ByteBuffer[] bufs;
+  private boolean closed = false;
+
+  private long position = 0;
+
+  public ECBlockReconstructedInputStream(ECReplicationConfig repConfig,
+      ECBlockReconstructedStripeInputStream stripeReader) {
```

Comment:
Why do we need to take stripereader from out side. I thought this class will abstract all Stripe reading related. I agree to take as arg if we have plans to plugin different type of impls.

Reply:
There will be a further class at the top level that abstracts the creation of this class + stripereader or ECBlockInputStream and also handles failures from ECBlockInputStream.

In general, I don't like to create complex objects (that need many parameters) from inside another class - injecting the dependencies is more flexible (eg we can inject a mock in tests easily, if we needed to). When I have the top level class ready, it will create this pair of objects together, perhaps using a factory (I am not sure on that yet).

Reply:
ok make sense.